src/pyipoptcoremodule.c

```diff
@@ -551,8 +555,39 @@ PyObject *set_intermediate_callback(PyObject * self, PyObject * args)
 	}
 }
 
-PyObject *solve(PyObject * self, PyObject * args)
+// too much cut and paste ... 
+#define SOLVE_CLEANUP() {  \
+  /* clean up and return */ \
+  if (retval == NULL) {   \
+    Py_XDECREF(x);        \
+    Py_XDECREF(mL);       \
+    Py_XDECREF(mU);       \
+    Py_XDECREF(lambda);   \
+  }                       \
+  SAFE_FREE(newx0);       \
+  return retval;          \
+  }
+
+#define SOLVE_CLEANUP_NULL() { \
+  retval = NULL;               \
+  SOLVE_CLEANUP()              \
+}
+
+#define SOLVE_CLEANUP_MEMORY() { \
+  retval = PyErr_NoMemory();     \
+  SOLVE_CLEANUP()                \
+}
+
+#define SOLVE_CLEANUP_TYPE(err) {          \
+  PyErr_SetString(PyExc_TypeError, err);   \
+  retval = NULL;                           \
+  SOLVE_CLEANUP()                          \
+}
+
+PyObject *solve(PyObject * self, PyObject * args, PyObject * keywds)
 {
+  static char *kwlist[] = {"x0", "userdata", "mult_g", "mult_x_L", "mult_x_U", NULL};
```

Comment:
I see. there is a subtle API change here (from args to kwargs), let me actually play with it a bit before saying it is good or bad.